packages/govern-subgraph/src/OptimisticQueue.ts

```diff
@@ -9,210 +9,311 @@ import {
   Revoked as RevokedEvent,
   Scheduled as ScheduledEvent,
   Vetoed as VetoedEvent,
-} from '../generated/templates/OptimisticQueue/OptimisticQueue'
+  Ruled as RuledEvent,
+  EvidenceSubmitted as EvidenceSubmittedEvent,
+  OptimisticQueue as OptimisticQueueContract,
+} from "../generated/templates/OptimisticQueue/OptimisticQueue";
 import {
-  Action,
-  Config,
-  Container,
-  Collateral,
-  OptimisticQueue,
-  Payload,
-  Role,
-} from '../generated/schema'
-
-const APPROVED_TYPE = 'Approved'
-const CANCELLED_TYPE = 'Cancelled'
-const CHALLENGED_TYPE = 'Challenged'
-const EXECUTED_TYPE = 'Executed'
-const SCHEDULED_TYPE = 'Scheduled'
-const VETOED_TYPE = 'Vetoed'
-
-const FREEZE_ADDR = '0x0000000000000000000000000000000000000001'
+  Config as ConfigEntity,
+  Challenge as ChallengeEntity,
+  Collateral as CollateralEntity,
+  Evidence as EvidenceEntity,
+  Item as ItemEntity,
+  OptimisticQueue as OptimisticQueueEntity,
+  Veto as VetoEntity,
+} from "../generated/schema";
+import { frozenRoles, roleGranted, roleRevoked } from "./lib/MiniACL";
+
+const NONE_STATUS = "None";
+const APPROVED_STATUS = "Approved";
+const CANCELLED_STATUS = "Cancelled";
+const CHALLENGED_STATUS = "Challenged";
+const EXECUTED_STATUS = "Executed";
+const REJECTED_STATUS = "Rejected";
+const SCHEDULED_STATUS = "Scheduled";
+const VETOED_STATUS = "Vetoed";
+
+const ALLOW_RULING = BigInt.fromI32(4);
+
+export function handleScheduled(event: ScheduledEvent): void {
+  const queue = loadOrCreateQueue(event.address);
+
+  const item = loadOrCreateItem(event.params.containerHash, event);
+
+  const scheduleDeposit = loadOrCreateCollateral(event, "1");
+  scheduleDeposit.token = event.params.collateral.token;
+  scheduleDeposit.amount = event.params.collateral.amount;
+
+  item.status = SCHEDULED_STATUS;
+  item.nonce = event.params.payload.nonce;
+  item.executionTime = event.params.payload.executionTime;
+  item.submitter = event.params.payload.submitter;
+  item.executor = event.params.payload.executor.toHexString();
+  item.proof = event.params.payload.proof;
+  item.collateral = scheduleDeposit.id;
+
+  // add the item
+  const currentItem = queue.queue;
+  currentItem.push(item.id);
+  queue.queue = currentItem;
+
+  item.save();
+  queue.save();
+}
+
+export function handleExecuted(event: ExecutedEvent): void {
+  // TODO: (Gabi) decide if we want to handle executinons here
+
+  const item = loadOrCreateItem(event.params.containerHash, event);
+
+  item.status = EXECUTED_STATUS;
+
+  item.save();
+}
 
 export function handleChallenged(event: ChallengedEvent): void {
-  const container = Container.load(event.params.containerHash.toHexString())
-  container.executionState = CHALLENGED_TYPE
-  container.save()
+  const item = loadOrCreateItem(event.params.containerHash, event);
+
+  item.status = CHALLENGED_STATUS;
+
+  const challenge = loadOrCreateChallenge(item.id, event);
```

Comment:
I include the basic information of a challenge but we probably want to sync with @facuspagnuolo to make sure we are providing everything we need to interact with AP.